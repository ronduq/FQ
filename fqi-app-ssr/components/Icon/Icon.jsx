import React from 'react';
import PropTypes from 'prop-types';

const Icon = ({ iconName, className}) => (
  <svg className={className} aria-hidden="true" focusable="false">
    <use xlinkHref={`#${iconName}`} />
  </svg>
);

Icon.propTypes = {
  className: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
};

export default Icon;