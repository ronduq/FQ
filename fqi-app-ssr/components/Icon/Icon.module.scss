@import '../../style/settings/colours';

// Icon sizes
.iconXSmall {
  height: 10px;
  width: 10px;
}

.iconSmall {
  height: 18px;
  width: 18px;
}

.iconMedium {
  height: 22px;
  width: 22px;
}

.iconLarge {
  height: 32px;
  width: 32px;
}

.iconHuge {
  height: 115px;
  width: 115px;
}

// Icon colours
.iconTeal {
  fill: $colour-main;
}

.iconOrange {
  stroke: $colour-secondary;
}

.iconOrangeFill {
  fill: $colour-secondary;
}

.iconBlack {
  stroke: $colour-grey-dark;
}