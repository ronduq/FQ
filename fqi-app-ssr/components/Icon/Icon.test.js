/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Icon from './Icon'

describe('With Snapshot Testing', () => {
  const props = {iconName:'test', className:'test'}
  it('Icon snapshot"', () => {
    const component = renderer.create(<Icon { ...props } />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });
})