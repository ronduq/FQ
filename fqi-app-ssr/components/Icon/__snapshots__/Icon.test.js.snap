// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Icon snapshot" 1`] = `
<svg
  aria-hidden="true"
  className="test"
  focusable="false"
>
  <use
    xlinkHref="#test"
  />
</svg>
`;
